projects/kalidea/kaligraphi/src/lib/molecules/kal-list/kal-list.component.ts

```diff
@@ -309,9 +330,17 @@ export class KalListComponent<T extends { id: string }> implements CollectionVie
     }
   }
 
-  ngOnInit() {
-    this.results = [];
+  destroySubscription() {
+    this.subscription.unsubscribe();
+
+    if (this.dataSource) {
```

Comment:
` if (this.dataSource && (this.dataSource as DataSource<T>).connect instanceof Function) {`